bundles/org.openhab.binding.touchwand/src/main/java/org/openhab/binding/touchwand/internal/dto/TouchWandAlarmSensorCurrentStatus.java

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.openhab.binding.touchwand.internal.dto;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+/**
+ * The {@link TouchWandAlarmSensorCurrentStatus} implements Alarm Sensor unit
+ * CurrentStatus data property.
+ *
+ * @author Roie Geron - Initial contribution
+ */
+
+@NonNullByDefault
+public class TouchWandAlarmSensorCurrentStatus {
+
+    private int batt;
+    private List<Sensor> sensorsStatus = new ArrayList<Sensor>();
+    private List<AlarmEvent> alarmsStatus = new ArrayList<AlarmEvent>();
+    private List<BinarySensorEvent> bSensorsStatus = new ArrayList<BinarySensorEvent>();
+
+    public void setBatt(Integer batt) {
+        this.batt = batt;
+    }
+
+    public int getBatt() {
+        return batt;
+    }
+
+    public void setSensorsStatus(List<Sensor> sensorsStatus) {
+        this.sensorsStatus = sensorsStatus;
+    }
+
+    public List<Sensor> getSensorsStatus() {
+        return sensorsStatus;
+    }
+
+    public List<BinarySensorEvent> getbSensorsStatus() {
+        return bSensorsStatus;
+    }
+
+    public void setbSensorsStatus(List<BinarySensorEvent> bSensorsStatus) {
+        this.bSensorsStatus = bSensorsStatus;
+    }
+
+    public List<AlarmEvent> getAlarmsStatus() {
+        return alarmsStatus;
+    }
+
+    public void setAlarmsStatus(List<AlarmEvent> alarmsStatus) {
+        this.alarmsStatus = alarmsStatus;
+    }
+
+    public class Alarm {
+        public int event;
+        public long ts;
+    }
+
+    public class AlarmEvent {
+        int alarmType;
+        Alarm alarm = new Alarm();
+    }
+
+    public class Sensor {
+        public int type;
+        public float value;
+    }
+
+    public class BinarySensor {
+        public long ts;
+        public boolean state;
+    }
+
+    public class BinarySensorEvent {
+        public int sensorType;
+        public BinarySensor sensor = new BinarySensor();
+    }
```

Comment:
Please make these nested classes static.
```suggestion
    public static class Alarm {
        public int event;
        public long ts;
    }

    public static class AlarmEvent {
        int alarmType;
        Alarm alarm = new Alarm();
    }

    public static class Sensor {
        public int type;
        public float value;
    }

    public static class BinarySensor {
        public long ts;
        public boolean state;
    }

    public static class BinarySensorEvent {
        public int sensorType;
        public BinarySensor sensor = new BinarySensor();
    }
```

Reply:
I would like to keep those non static since I need to create new instances of Sensors , Binary sensors end add it to the sensors status ArrayList during Json deserialization [for example](https://github.com/roieg/openhab-addons/blob/ba5c54d2dfda06b89d6c0d11a6ed0e45ba9712b1/bundles/org.openhab.binding.touchwand/src/main/java/org/openhab/binding/touchwand/internal/dto/AlarmSensorUnitDataDeserializer.java#L95)
Changing the inner classes to static will require using adding access methods in the outer class (which is possible but not sure if it is needed)

Reply:
You can create static instances like `new TouchWandAlarmSensorCurrentStatus.AlarmEvent()`.

Reply:
Thanks , done